### PR TITLE
docs: fixed typos and grammatical errors

### PR DESCRIPTION
**Pull request by Villy-P**
#### Description of Change

Fixed typos and grammatical mistakes within the documentation, along with changing certain words to American English.

#### Checklist

- [x] `npm test` passes
- [x] relevant documentation, tutorials, templates and examples are changed or added

#### Release Notes

notes: none

### CODE REVIEW

File: docs/tutorial/performance.md [hunk 1]
```diff
@@ -62,14 +62,34 @@ at once, consider the [Chrome Tracing](https://www.chromium.org/developers/how-t
...
+      - [How?](#how-4)
+    - [6. Unnecessary or blocking network requests](#6-unnecessary-or-blocking-network-requests)
+      - [Why?](#why-5)
+      - [How?](#how-5)
+    - [7. Bundle your code](#7-bundle-your-code)
+      - [Why?](#why-6)
+      - [How?](#how-6)
+    - [8. Call `Menu.setApplicationMenu(null)` when you do not need a default menu](#8-call-menusetapplicationmenunull-when-you-do-not-need-a-default-menu)
+      - [Why?](#why-7)
+      - [How?](#how-7)
```
Comment (VerteDinde) [inline docs/tutorial/performance.md:92]: Appreciate the fixes, but we actually don't want to include the Performance header and items like Why/How in this list as it makes it too verbose. Could you please revert this part to how it was previously?

File: docs/tutorial/performance.md [hunk 2]
```diff
@@ -83,7 +103,7 @@ is not in fact the leanest or smallest one available.
 
 The reasoning behind this recommendation is best illustrated with a real-world
 example. During the early days of Electron, reliable detection of network
-connectivity was a problem, resulting many apps to use a module that exposed a
+connectivity was a problem, resulting in many apps to use a module that exposed a
```
Comment (VerteDinde) [inline docs/tutorial/performance.md:106]: I think this is still incorrect:
```suggestion
connectivity was a problem, resulting in many apps using a module that exposed a
```

**Review by VerteDinde — CHANGES_REQUESTED** [review]
Thanks for the PR! Two comments, but we can merge with those are addressed

**Review by ckerr — APPROVED** [review]
Thanks!

**Review by jkleinsc — APPROVED** [review]

File: docs/tutorial/sandbox.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ further configuration. If you want to disable the sandbox for a process, see the
 [Disabling the sandbox for a single process](#disabling-the-sandbox-for-a-single-process)
 section.
 
-## Sandbox behaviour in Electron
+## Sandbox behavior in Electron
```
Comment (codebytere) [inline docs/tutorial/sandbox.md:20]: This is not an incorrect spelling - it's British English. We can change this given the majority of our documentation follows American English but strictly speaking it's not a typo.

Comment (Villy-P) [reply docs/tutorial/sandbox.md:20]: I have changed the description of the PR. Thank you!

**Review by VerteDinde — APPROVED** [review]
Thanks for the PR! 🙂

**Pull request merged**